src/common/legacy_config_opts.h

```diff
@@ -1509,6 +1509,8 @@ OPTION(rgw_crypt_vault_addr, OPT_STR) // Vault server base address
 OPTION(rgw_crypt_vault_prefix, OPT_STR) // Optional URL prefix to Vault secret path
 OPTION(rgw_crypt_vault_secret_engine, OPT_STR) // kv, transit or other supported secret engines
 OPTION(rgw_crypt_vault_namespace, OPT_STR) // Vault Namespace (only availabe in Vault Enterprise Version)
+OPTION(rgw_crypt_vault_verify_ssl, OPT_BOOL) // should we try to verify vault's ssl
```

Comment:
It would be nice to support client cert and client private key as well.

Reply:
@leseb : The second commit have option for vault client [ca cert](https://github.com/ceph/ceph/pull/37730/commits/491997b66b6c29407947d2ea923c44f174e8a45f). That will be enough for the tls authentication , Am I missing something?

Reply:
Ok for the cert, what about the private key?

Reply:
Either I can use separate option for the private key using [CURLOPT_SSH_PRIVATE_KEYFILE](https://curl.se/libcurl/c/CURLOPT_SSH_PRIVATE_KEYFILE.html) or we can point to directory which contains bundle [CURLOPT_CAPATH](https://curl.se/libcurl/c/CURLOPT_CAPATH.html)

Reply:
@leseb any suggestions?

Reply:
The "verify_ssl" option should default to true.  The documentation should make it clear that setting it "false" is not recommended, dangerous, and nearly always avoidable.

The client key/cert stuff shoud be part of a separate PR - and it has to include TTL / periodic renewal logic to be useful.  Note that vault agent can already do that, so this feature may not be necessary.

Reply:
I have updated PR with suggested changes and client key/cert will address as separate PR